src/skeleplex/measurements/branches.py

```diff
@@ -0,0 +1,472 @@
+
+            label_slice = ski.measure.label((segmentation_slice != 0).astype(np.uint8))
+            props = ski.measure.regionprops(label_slice)
+
+            if props:
+                minor_axis = props[0].minor_axis_length
+                major_axis = props[0].major_axis_length
+                total_area = props[0].area
+                minor_axis_branch.append(minor_axis)
+                major_axis_branch.append(major_axis)
+                total_area_branch.append(total_area)
+
+            if np.sum(segmentation_slice == 1) > 0:
+                lumen_label = (segmentation_slice == 2).astype(np.uint8)
+                tissue_label = (segmentation_slice == 1).astype(np.uint8)
+
+                lumen_props = ski.measure.regionprops(ski.measure.label(lumen_label))
+                tissue_props = ski.measure.regionprops(ski.measure.label(tissue_label))
+
+                if lumen_props and tissue_props:
+                    lumen_area = lumen_props[0].area
+                    tissue_area = tissue_props[0].area
+                    total_area = lumen_area + tissue_area
+                    total_radius = radius_from_area(total_area)
+                    lumen_radius = radius_from_area(lumen_area)
+                    tissue_radius = total_radius - lumen_radius
+                    tissue_radius_branch.append(tissue_radius)
+                    lumen_radius_branch.append(lumen_radius)
+            else:
+                # no tissue label, full region is tissue
+                tissue_radius_branch.append(minor_axis / 2)
+                lumen_radius_branch.append(0)
+        else:
+            # completely closed (no lumen)
+            label_slice = ski.measure.label((segmentation_slice != 0).astype(np.uint8))
+            props = ski.measure.regionprops(label_slice)
+            if props:
+                minor_axis = props[0].minor_axis_length
+                major_axis = props[0].major_axis_length
+                total_area = props[0].area
+                minor_axis_branch.append(minor_axis)
+                major_axis_branch.append(major_axis)
+                total_area_branch.append(total_area)
+                tissue_radius_branch.append(minor_axis / 2)
+                lumen_radius_branch.append(0)
+
+    return (
+        start_node,
+        end_node,
+        np.mean(np.array(lumen_radius_branch) * 2),
+        np.std(np.array(lumen_radius_branch) * 2),
+        np.mean(tissue_radius_branch),
+        np.std(tissue_radius_branch),
+        np.mean(total_area_branch),
+        np.std(total_area_branch),
+        np.mean(minor_axis_branch),
+        np.std(minor_axis_branch),
+        np.mean(major_axis_branch),
+        np.std(major_axis_branch),
+    )
+
+
+def add_measurements_from_h5_to_graph(graph_path, input_path):
+    """
+    Add measurements from HDF5 files to the skeleton graph.
+
+    The slice names need to be in the format:
+
+    {base}_{name}_{start_node}_{end_node}.h5
```

Comment:
Please briefly explain where the h5 files should come from (i.e., which functions create valid files).